tls/s2n_kem.c

```diff
@@ -101,6 +101,33 @@ const struct s2n_kem_group *ALL_SUPPORTED_KEM_GROUPS[S2N_SUPPORTED_KEM_GROUPS_CO
  * functions may return any non-zero value to indicate failure. */
 #define GUARD_PQ_AS_RESULT(x) RESULT_ENSURE((x) == 0, S2N_ERR_PQ_CRYPTO)
 
+/* Checks if the Hybrid KEM is using the old Hybrid KEM format (PQ Length, PQ Share, ECC Length, ECC Share) or the new
+ * format (PQ Share, ECC Share). The new format will always be 4 bytes shorter than the old format, so we check if the
+ * total length is 4 bytes longer than expected for the new format, and if so, fall back to the old format. */
+int s2n_is_hybrid_kem_length_prefixed(const s2n_mode mode, uint16_t actual_hybrid_share_size, const struct s2n_kem_group *kem_group, bool *is_length_prefixed)
+{
+    uint16_t unprefixed_hybrid_share_size = 0;
+
+    if (mode == S2N_CLIENT) {
+        unprefixed_hybrid_share_size = kem_group->curve->share_size + kem_group->kem->public_key_length;
+    } else {
+        unprefixed_hybrid_share_size = kem_group->curve->share_size + kem_group->kem->ciphertext_length;
+    }
+
+    /* Initial draft versions of Hybrid PQ TLS spec required that each individual key share be prefixed by it's length.
```

Comment:
Do you have any idea how close the draft is to being finalized? This is would be annoying to redo if they change the format again.

Reply:
This is the first breaking change that I'm aware of in PQ TLS 1.3 draft RFC. It was first published in March 2019, so 4 years of being stable so far. I doubt that there would be another breaking change to how Hybrid KEM's should be formatted and wrapped, but it's possible. Estimates are that the PQ TLS 1.3 RFC will likely be finalized and published within the next 2-ish years. 

Kyber itself is likely to be tweaked one last time during the final standardization process, but that updated version of Kyber will be assigned a new algorithm ID and won't be a breaking change to the higher-level wrapping format like this draft RFC change was.

Reply:
Note: the current draft expires on 1 March 2023. To save ourselves some work and heartache, is it worth holding off on pushing these changes until a new revision is published?

Reply:
Draft RFC's expire after 6 months if no new revisions are published as far as I'm aware. So it's usually expected that there will be a new revision of a draft RFC a few months away. Most new revisions make some small wording changes and clarifications, but don't change the actual protocol, so I'm not worried about a new draft coming out and us needing to make changes.

Reply:
Sure. I guess my point is that the authors are going to have to release a new revision within the next ~2 weeks. I agree that it's very unlikely that it will not contain any changes to the protocol itself. But waiting 2 weeks to merge these changes seems like a pretty small cost to ensure that we aren't immediately incompatible with the protocol if there are substantive changes.